spec/non_default_api_spec.rb

```diff
@@ -497,7 +497,9 @@ def app
   end
 
   context ':hide_format' do
-    before :all do
+    let(:doc_options) { Hash.new }
+    before do
+      stub_const('DOC_OPTIONS', doc_options)
```

Comment:
What I tried achieving here is providing different options for `add_swagger_documentation` in `let` blocks, while still having one setup code for all specs (see `doc_options`).

Because the method is invoked inside the class definition, the only way I could pass something to it is via a defining a constant in a scope where the let's can be accessed, then in turn the constant is available in the class definition.

It's a bit hacky, I would love for a way to achieve the same without resorting to trickery :-)